File: cwt-spring/src/main/java/com/cwtsite/cwt/domain/game/service/RatingRepository.java
package com.cwtsite.cwt.domain.game.service;

import com.cwtsite.cwt.domain.game.entity.Rating;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface RatingRepository extends JpaRepository<Rating, Long> {
}

File: cwt-spring/src/main/java/com/cwtsite/cwt/domain/game/service/RatingRepository.kt
package com.cwtsite.cwt.domain.game.service

import com.cwtsite.cwt.domain.game.entity.Game
import com.cwtsite.cwt.domain.game.entity.Rating
import com.cwtsite.cwt.domain.game.entity.enumeration.RatingType
import com.cwtsite.cwt.domain.user.repository.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface RatingRepository : JpaRepository<Rating, Long> {

    fun findByUserAndGameAndTypeIn(user: User, game: Game, type: List<RatingType>): List<Rating>
}
